mongo/job_store.go

```diff
@@ -98,14 +98,18 @@ func (m *JobStore) CreateJob(ctx context.Context, id string) (models.Job, error)
 }
 
 // CreateTask creates a new task, for the given API and job ID, in the collection, and assigns default values to its attributes
-func (m *JobStore) CreateTask(ctx context.Context, jobID string, nameOfApi string, numDocuments int) (models.Task, error) {
-	log.Info(ctx, "creating task in mongo DB", log.Data{"jobID": jobID, "nameOfApi": nameOfApi, "numDocuments": numDocuments})
+func (m *JobStore) CreateTask(ctx context.Context, jobID string, taskName string, numDocuments int) (models.Task, error) {
+	log.Info(ctx, "creating task in mongo DB", log.Data{"jobID": jobID, "taskName": taskName, "numDocuments": numDocuments})
 
 	// If an empty job id was passed in, return an error with a message.
 	if jobID == "" {
 		return models.Task{}, ErrEmptyIDProvided
 	}
 
+	if taskName == "" {
```

Comment:
This would be better handled inside the handler, not the mongo store.

So inside `CreateTaskHandler` handler:

```
    if err := task.Validate(); err != nil {
        log.Error(ctx, "CreateTask endpoint: Invalid request body", err)
        http.Error(w, invalidBodyErrorMessage, http.StatusBadRequest)
    }
```

where in models/task.go:

```
func (task *Task) Validate() error {
	if task.TaskName == {
        	return errs.ErrEmptyTaskNameProvided
        }
}
```

where errs is the apierrors package and inside apierrors.go add `var ErrEmptyTaskNameProvided = errors.New("task name must not be an empty string")`